src/modules/bootloader/main.py

```diff
@@ -605,17 +605,71 @@ def run_grub_install(fw_type, partitions, efi_directory):
+            check_target_env_call(["sh", "-c", "/usr/bin/find /boot/ -maxdepth 1 -name 'vmlinuz-*' | /usr/bin/xargs -I{} /bin/sh -c 'if ! /usr/bin/sbverify --cert /etc/secureboot/MOK.crt {} >/dev/null 2>&1 ;then sudo /usr/bin/sbsign --key " + MOK_path + "MOK.key --cert " + MOK_path + "MOK.crt --output {} {}; fi'"])
+            check_target_env_call(["sh", "-c", "/usr/bin/sudo /usr/bin/sbsign --key " + MOK_path + "MOK.key --cert " + MOK_path + "MOK.crt --output " + grub_path + " " + grub_path])
+            check_target_env_call(["sh", "-c", "/usr/bin/sudo /usr/bin/mokutil --import " + MOK_path + "MOK.cer --root-pw"])
+            install_efi_directory = install_path + efi_directory
+            shutil.copy2(os.path.join(install_path, "/usr/share/shim-signed/shimx64.efi") , os.path.join(install_efi_directory, "EFI", efi_bootloader_id))
+            shutil.copy2(os.path.join(install_path, "/usr/share/shim-signed/mmx64.efi") , os.path.join(install_efi_directory, "EFI", efi_bootloader_id))
+            shutil.copy2(os.path.join(install_path, "/usr/share/shim-signed/shimx64.efi") , os.path.join(install_efi_directory, "EFI", efi_bootloader_id, "bootx64.efi"))
+            efi_drive = subprocess.check_output([
+            libcalamares.job.configuration["grubProbe"],
+            "-t", "drive", install_efi_directory]).decode("ascii")
+            efi_disk = subprocess.check_output([
+            libcalamares.job.configuration["grubProbe"],
+            "-t", "disk", install_efi_directory]).decode("ascii")
+            efi_drive_partition = efi_drive.replace("(","").replace(")","").split('/')[3]
+            # Get the first run of digits from the partition
+            efi_partition_number = None
+            c = 0
+            start = None
+            while c < len(efi_drive_partition):
+                if efi_drive_partition[c].isdigit() and start is None:
+                    start = c
+                if not efi_drive_partition[c].isdigit() and start is not None:
+                    efi_partition_number = efi_drive_partition[start:c]
+                    break
+                c += 1
+            if efi_partition_number is None:
+                raise ValueError("No partition number found for %s" % install_efi_directory)
+
+            install_efi_bin = "shimx64.efi"
+            subprocess.call([
+                libcalamares.job.configuration["efiBootMgr"],
+                "-c",
+                "-w",
+                "-L", efi_bootloader_id + "Shim",
+                "-d", efi_disk,
+                "-p", efi_partition_number,
+                "-l", os.path.join(install_efi_directory, "EFI", efi_bootloader_id, install_efi_bin)])
+
+            efi_boot_next()
+
+            check_target_env_call([libcalamares.job.configuration["grubMkconfig"],
+                           "-o", os.path.join(efi_directory, "EFI",
+                                              efi_bootloader_id, "grub.cfg")])
```

Comment:
`grub-mkconfig` is already being run after this function ends.  Is this needed in addition to that?